python/tk_desktop/install_apps.py

```diff
@@ -0,0 +1,250 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import pprint
+
+import sgtk
+
+from sgtk.platform.qt import QtCore
+from sgtk.platform.qt import QtGui
+
+from .ui import install_apps
+
+shotgun_data = sgtk.platform.import_framework("tk-framework-shotgunutils", "shotgun_data")
+
+class InstallApps(QtGui.QWidget):
+    """
+    Widget displayed in the desktop Window's project_commands wiget when no
+    "button" commands are found for the current project. Displays site
+    Software entity icons and a link to a Shotgun Support article detailing
+    how to configure Software entities.
+    """
+    def __init__(self, parent=None):
+        QtGui.QWidget.__init__(self, parent)
+
+        self.ui = install_apps.Ui_InstallApps()
+        self.ui.setupUi(self)
+
+        # Assign magic numbers and such
+        self._icon_size = QtCore.QSize(64, 64)
+        self._icon_layout_spacing = 5
+        self._icon_layout_stretch = 1
+        self._icon_column_count = 4
+        self._config_link_color = "#0000FF"
+        self._config_link_text = "here"
+
+        # Hide this widget by default.
+        self.setVisible(False)
+
+    def build(self, project=None):
+        """
+        Query Shotgun for a list of Software images to display and
+        set the Software configuration link value
+
+        :param project: Current Shotgun Project instance (dict)
+        """
+        # Make sure we can get a handle to the current engine.
+        # This is primarily used for logging purposes.
+        engine = sgtk.platform.current_engine()
```

Comment:
Could you explain when this is possible? Since that piece of code is executed when the engine runs, I have a hard time understanding when this would happen. And if I have a hard time understanding, then it needs a comment. :p